text/0000-path-clarity.md

```diff
@@ -0,0 +1,734 @@
+## As we would teach it
+
+The following sections sketch a plausible way of teaching the module system once
+this RFC has been fully implemented.
+
+### Using external dependencies
+
+To add an external dependency, record it in the `[dependencies]` section of
+`Cargo.toml`:
+
+```toml
+[dependencies]
+serde = "1.0.0"
+```
+
+By default, crates have an automatic dependency on `std`, the standard library.
+
+Once your dependency has been added, you can bring it or its exports into scope with
+`use` declarations:
+
+```rust
+use std; // bring `std` itself into scope
+use std::vec::Vec;
+
+use serde::Serialize;
+```
+
+Note that these `use` declarations all begin with a crate name.
+
+Once an item is in scope, you can reference it directly within definitions:
+
+```rust
+// Either of these work, because we brought `std` into scope:
+fn make_vec() -> Vec<u8> { ... }
+fn make_vec() -> std::vec::Vec<u8> { ... }
+
+// Only the first of these work, because we didn't bring `serde` into scope:
+impl Serialize for MyType { ... }
+impl serde::Serialize for MyType { ... } // the name `serde` is not in scope here
+```
+
+You can also reference items from a crate without bringing them into scope by
+writing a **fully qualified path**, designated by a leading `::`, as follows:
+
+```rust
+impl ::serde::Serialize for MyType { ... }
+```
+
+All `use` declarations are interpreted as fully qualified paths, making the
+leading `::` optional for them.
+
+> **Note: that means that you can write `use serde::Serialize` in *any* module
+without trouble, as long as `serde` is an external dependency!**
+
+### Adding a new file to your crate
+
+Rust crates have a distinguished entry point (generally called `main.rs` or
+`lib.rs`) which is used to determine the crate's structure. Other files and
+directories within `src/` are *not* automatically included in the crate.
+Instead, you explicitly declare *submodules* using `mod` declarations.
+
+Let's see how this looks with an example. First, we might set up a directory
+structure like the following:
+
+```
+src
+├── cli
+│   ├── parse.rs
+│   └── usage.rs
+├── cli.rs
+├── main.rs
+├── process
+│   ├── read.rs
+│   └── write.rs
+└── process.rs
+```
+
+The intent is for the crate to have two top-level modules, `cli` and `process`,
+each of which contain two submodules. To turn these files into submodules, we
+use `mod` declarations as follows:
+
+```rust
+// src/main.rs
+mod cli;
+mod process;
+```
+
+```rust
+// src/cli.rs
+mod parse;
+mod usage;
+```
+
+```rust
+// src/process.rs
+mod read;
+mod write;
+```
+
+Note how these declarations follow the structure of the filesystem (except that
+the entry point, `main.rs`, has its children modules as sibling files). By
+default, `mod` declarations assume this kind of direct mapping to the
+filesystem; they are used to tell Rust to incorporate those files, and to set
+attributes on the resulting modules (as we'll see in a moment).
+
+### Importing items from other parts of your crate
```

Comment:
Nit: this section doesn't make it clear that `crate` and `crate::` are two different things, which might make it confusing to people who haven't been following the discussion...